sdk/go/kubernetes/helm/v2/chart.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// *** WARNING: this file was generated by pulumigen. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package helm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/yaml"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+// Chart is a component representing a collection of resources described by an arbitrary Helm
+// Chart. The Chart can be fetched from any source that is accessible to the `helm` command
+// line. Values in the `values.yml` file can be overridden using `ChartOpts.values` (equivalent
+// to `--set` or having multiple `values.yml` files). Objects can be transformed arbitrarily by
+// supplying callbacks to `ChartOpts.transformations`.
+//
+// `Chart` does not use Tiller. The Chart specified is copied and expanded locally; the semantics
+// are equivalent to running `helm template` and then using Pulumi to manage the resulting YAML
+// manifests. Any values that would be retrieved in-cluster are assigned fake values, and
+// none of Tiller's server-side validity testing is executed.
+type Chart struct {
+	pulumi.ResourceState
+
+	Resources pulumi.Output
+}
+
+// NewChart registers a new resource with the given unique name, arguments, and options.
+func NewChart(ctx *pulumi.Context,
+	name string, args ChartArgs, opts ...pulumi.ResourceOption) (*Chart, error) {
+
+	// Register the resulting resource state.
+	chart := &Chart{}
+	err := ctx.RegisterComponentResource("kubernetes:helm.sh/v2:Chart", name, chart, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make the component the parent of all subsequent resources.
+	opts = append(opts, pulumi.Parent(chart))
```

Comment:
This doesn't seem to match other SDKs (or be right).  the opts apply to the chart, not necessarily to it's children.  We just want to pass `Parent` into the children.

Reply:
Oops, copied this from an earlier version of ConfigFile